src/Units/Unit.java

```diff
@@ -185,43 +191,65 @@ public void fire(WeaponModel weapon)
     }
   }
 
-  public int getHP()
+  /** @return unit HP * 10 */
+  public int getEffectiveHealth()
+  {
+    return effectiveHealth(health);
+  }
+  /** @return unit actual health */
+  public int getPreciseHealth()
   {
-    return (int) Math.ceil(healthToHP(health));
+    return health;
   }
-  public double getPreciseHP()
+  public int getHP()
   {
     return healthToHP(health);
   }
-  private static double healthToHP(int input)
+  /** @return health value == its corresponding HP value * 10 */
+  public static int effectiveHealth(int input)
```

Comment:
This... is not quite what I envisioned when I suggested keeping the units consistent everywhere except when displaying the value.

Reply:
The problem is that HP is *actually* more important most of the time.
Attack/defense scaling, healing, and anything to with funds don't actually interact with the actual health mechanic.

Reply:
Can we not adjust those calculations to work with the [0-100] health values instead?
I _really_ don't like the idea of having HP =/= health.

Reply:
Technically, yes, but practically no.
Specifically, attack/defense/etc all work on [0-10], rounded up. So either we provide an accessor on Unit to get that value (if nothing else, doing the rounding up), then we *have* to do that calculation manually everywhere else.
Also worth noting is that when I think about HP in CO design/etc, I'm thinking about [1-10]. I don't want to remember/care about there being a precise health concept; it exists almost entirely to broker when fractional bits of damage should add up to a full HP.
If we do provide that accessor, we need good terminology to separate the two concepts or we (or at least I) will get constantly confused. ("effective health" IMO isn't good enough, at least not without *also* linking it to HP)

Reply:
> Technically, yes, but practically no

Technically===practically. 

> attack/defense/etc all work on [0-10]

Then these are the calculations I'm suggesting we modify; your answer seems to presuppose that we can't do that for some reason.

> So either we provide an accessor on Unit to get that value

We do not. We only hand out the One True [0-100] value, and the UI divides by 10 and ceilings the result before displaying.
(Also you didn't provide an _or_ option here)

> Also worth noting is that when I think about HP in CO design/etc, I'm thinking about [1-10].

Yes; that was the one downside of this approach I mentioned previously. I'm willing to live with that though if it makes life easier in other ways. Weapon-damage values etc are already stored this way, so it seems like a sensible/straightforward thing to do.

Reply:
> > Technically, yes, but practically no
> 
> Technically===practically.

That's... literally not true, but I think I can see what you're going for.

> > attack/defense/etc all work on [0-10]
> 
> Then these are the calculations I'm suggesting we modify; your answer seems to presuppose that we can't do that for some reason.

That's cart behavior, and I subjectively prefer it greatly because it makes combat easier to reason about (Wargroove uses the exact health percentage, so I'm not just guessing when I say I like it better).
I'm not changing it. You can make a fork if you really care that much.

> > So either we provide an accessor on Unit to get that value
> 
> We do not. We only hand out the One True [0-100] value, and the UI divides by 10 and ceilings the result before displaying.

For the reasons mentioned above... I'm not gonna go for that.

> (Also you didn't provide an _or_ option here to match with your _either_)

...whoops. The "then" after the comma should have been the "or".

Reply:
> Wargroove uses the exact health percentage, so I'm not just guessing when I say I like it better

I wasn't suggesting to change any gameplay behavior. The functions themselves could stay exactly the same, since it's easy enough to pull the unit's health and `ceil(number/10)` before plugging it into the existing calculation. That should be a local implementation detail as part of the formula though, not a system-wide complexity magnifier.

Besides, if the carts store the HP as a [0-100] value, that's basically what they have to be doing anyway, right? Actually, it looks like you _are_ essentially doing that already. You get the health, convert it to HP, and then run the calculations. Cool. Well then. I guess we managed to disagree while saying the same thing here. Fantastic.

I'm just trying to avoid littering the `Unit` interface with excessive inscrutable accessors, and avoid multiple representations of the same value (DRY, etc).
Right now we have:
`getHP()`
`getHPFactor()`
`getPreciseHealth()`
`getEffectiveHealth()`
`effectiveHealth()` (static)
`healthToHP()`      (static)
`healthFromHP()` (static)

This is a bit much, and some of these are only used in one or two places. I made a suggestion in the other head, so I won't bother repeating it here.

Reply:
> I guess we managed to disagree while saying the same thing here. Fantastic.

You did say you didn't like HP and health not being the same thing, at least initially.
Fair enough, though.

Reply:
I don't like them not being the same thing. I would just call the down-conversion to HP an implementation detail of the combat calculations, and not give it a separate label at all (at least in the world where we decided [0-100] is the True Value).
But if we decide [0-10] is the Real Thing, then we can hide the [0-100] dealio inside `Unit` and call it whatever we want: `health`, `healthPercent`, `inverseClosenessToDying`, `nobodyActuallyLikesCheeseWhiz`, whatever.